java-checks/src/test/files/checks/ClassFieldCountCheck/UnusualTrees.java

```diff
@@ -0,0 +1,26 @@
+enum TooManyFieldsEnum { // Noncompliant [[sc=6;ec=23]] {{Refactor this class so it has no more than 2 fields, rather than the 3 it currently has.}}
+  ENUM_CONSTANT_TREE { // Noncompliant [[sc=3;ec=4]] {{Refactor this class so it has no more than 2 fields, rather than the 3 it currently has.}}
+     int field1;
+     int field2;
+     int field3;
+  };
+   int field1;
+   int field2;
+   int field3;
+}
+
+interface TooManyFieldsInterface { // Noncompliant [[sc=11;ec=33]] {{Refactor this class so it has no more than 2 fields, rather than the 3 it currently has.}}
+   int field1;
+   int field2;
+   int field3;
+}
+
+class TooManyFieldsAnonymous {
+  {
+    new java.util.ArrayList<Integer>() { // Noncompliant [[sc=5]] {{Refactor this class so it has no more than 2 fields, rather than the 3 it currently has.}}
```

Comment:
You should have checked for for the `end column`. Specifying `end line` with `el` can help.

Usually, we try to limit as much as possible large area being highlighted. In that specific case, highlighting everything, starting from the new keyword and including the whole body is too much. For anonymous classes, to only highlight the type tree is enough.